src/object/obj_utils.c

```diff
@@ -100,38 +98,53 @@ obj_query_merge_recx(struct daos_oclass_attr *oca, daos_unit_oid_t oid, daos_key
 	uint64_t	cell_rec_nr;
 
 	if (!daos_oclass_is_ec(oca))
-		D_GOTO(out, changed = true);
+		return false;
```

Comment:
Suggest to move "if (!daos_oclass_is_ec(oca))" into obj_query_reduce_recx(), then obj_query_fix_recx() only handles EC object without return value.

Reply:
done